client/src/templates/Challenges/odin/show.tsx

```diff
@@ -78,9 +79,9 @@ interface ShowOdinProps {
 interface ShowOdinState {
   subtitles: string;
   downloadURL: string | null;
-  selectedOption: number | null;
-  answer: number;
-  isWrongAnswer: boolean;
+  selectedMcqOptions: (number | null)[];
+  submittedMcqAnswers: (number | null)[];
```

Comment:
I am struggling to see a reason for such a weird type. That is, why not go with `number[]`?

Reply:
they start out as `null[]` with a length of the number of questions - I suppose undefined might be more appropriate. But yea, so the selected options are all unselected when you go to the challenge, then when you select an option, it changes the index of that selected option to the selection - the others are still null. Select an option for all the questions, and then there isn't any more nulls for the `selectedOptions` - click "check your answers" and those options get copied to the `submittedAnswers` - so we know what they last submitted and show the feedback. From there, they can change any answers and try to submit again. Maybe get it running and give it a try to see the UX if you haven't.

So, I guess we could do something like store all those as -1 to start instead of null? It feels like null or undefined is more appropriate. Any suggestions?

Reply:
Right, I understand what is going on. I just think it is odd to do that instead of storing an array of ids (indexes) of just the selected stuff.

Reply:
So, I think I need some clarification still. You don't know why the `submittedMcQAnswers` exists? or you don't like the type `selectedMcqOptions: (number | null)[];` - if it's the second, I'm not understanding how I could just store an arrays of id's. Can you give me an example of what that would look like? For some more clarification on my end, the array currently stores the selected option index in the array at the index of the question. So if you had three MCQ questions, the array might look like this: `[2, 1, null]` - meaning the currently selected option for question 1 is answer 2, question 2 is answer 1, and question 3 doesn't have a selected option yet.

The "check your answers" button is not disabled if an answer isn't selected, so you can submit something with `null` - perhaps that's not the way to go, but that's how it currently works. So I left it like that.

Reply:
Ah. That was a misunderstanding on my part. Thank you.

I suppose it does work to have it with that type.